src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -1,74 +1,68 @@
+import static frc.robot.subsystems.IntakeStates.States.*;
+
+import com.revrobotics.CANSparkMax;
 
 import edu.wpi.first.math.filter.Debouncer;
 import edu.wpi.first.networktables.BooleanPublisher;
 import edu.wpi.first.networktables.NetworkTable;
 import edu.wpi.first.networktables.NetworkTableInstance;
 import edu.wpi.first.networktables.PubSubOption;
+import edu.wpi.first.networktables.StringPublisher;
 import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Config;
+import frc.robot.subsystems.IntakeStates.Modes;
+import frc.robot.subsystems.IntakeStates.States;
 
-import com.revrobotics.CANSparkLowLevel.MotorType;
-
-
-public class IntakeSubsystem extends SubsystemBase {
+/** Add your docs here. */
+public class IntakeSubsystem extends SubsystemBase{
+    private CANSparkMax m_intake;
+    private IntakeStates intakeStates = new IntakeStates();
 
-    private DigitalInput sensor7; //DigitInput instance
-    private DigitalInput sensor8;
-    private DigitalInput sensor9;
+    private DigitalInput sensor7;//Back
+    private DigitalInput sensor8;//Center
+    private DigitalInput sensor9;//Front
 
-    Debouncer sensor7Debouncer;
-    Debouncer sensor8Debouncer;
-    Debouncer sensor9Debouncer;
+    private Debouncer sensor7Debouncer;
+    private Debouncer sensor8Debouncer;
+    private Debouncer sensor9Debouncer;
 
     private BooleanPublisher sensor7Pub;
     private BooleanPublisher sensor8Pub;
     private BooleanPublisher sensor9Pub;
+    private StringPublisher statesPub;
     
     private boolean sensor7Result;
     private boolean sensor8Result;
     private boolean sensor9Result;
 
-    private static final IntakeSubsystem INSTANCE_INTAKE = new IntakeSubsystem();
-
-    private CANSparkMax m_intake;
-    int targetSpeed = 0;
-    double kMaxOutput = 1;
-    double kMinOutput = -1;
-
-    public boolean isActive() {
-        return m_intake != null;
-    }
-
+    private static IntakeSubsystem instance;
     public static IntakeSubsystem getInstance() {
-        if (INSTANCE_INTAKE.isActive())
-            return INSTANCE_INTAKE;
-        else
-            return null;
+        if (instance == null)
+            instance = new IntakeSubsystem();
+        return instance;
     }
 
-
     private IntakeSubsystem() {
-        if (Config.Intake.INTAKE != -1) {
-            initializeSubsystem();
-        } else {
-            m_intake = null;
-        }
-    }
-
-    private void initializeSubsystem() {
-        
+        System.out.println("[Init]Creating Intake");
         m_intake = new CANSparkMax(Config.Intake.INTAKE, MotorType.kBrushless);
         m_intake.restoreFactoryDefaults();
         m_intake.setInverted(true);
         m_intake.setSmartCurrentLimit(70);
         m_intake.setIdleMode(IdleMode.kBrake);
+        m_intake.enableVoltageCompensation(6);//Check this out
```

Comment:
I'm fairly confident that setting the voltage compensation to 6 means the sparkmax will never apply more than 6 volts to the NEO. This conflicts with your line above that tries to use 7 volts to shoot.